hardware_interface/include/hardware_interface/joint.hpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__JOINT_HPP_
+#define HARDWARE_INTERFACE__JOINT_HPP_
+
+#include <string>
+#include <vector>
+
+#include "hardware_interface/hardware_and_component_info.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/visibility_control.h"
+
+namespace hardware_interface
+{
+
+class Joint
+{
+public:
+  HARDWARE_INTERFACE_PUBLIC
+  Joint() = default;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  ~Joint() = default;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  return_type configure(const ComponentInfo & joint_info) = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  std::vector<std::string> get_command_interfaces() const = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  std::vector<std::string> get_state_interfaces() const = 0;
+
+  HARDWARE_INTERFACE_EXPORT
+  virtual
+  return_type get_command(
+    std::vector<double> & command,
+    std::vector<std::string> & interfaces) const = 0;
```

Comment:
This needs doxygen to explain what's happening here.

Shouldn't `interfaces` be `const` here as it's only the identifier(s) for which interface to get the current command value? Also, does this make much sense to ask for a command value? The command `set_command` is triggered, the `get_command` value is already outdated, isn't it?

Reply:
Oh... sorry I missed this one... sorry. I will add doxygen description.

Reply:
I was also setting `interfaces` int he tests if empty list is provided.... I can also imagine this use-case..

Reply:
> I was also setting interfaces int he tests if empty list is provided

Sorry, I don't understand this. Do you mind explaining this once more? I can see how a `set_command` makes sense, but when would I call `get_command`?

Reply:
```suggestion
  HARDWARE_INTERFACE_EXPORT
  virtual
  return_type get_command(
    std::vector<double> & command,
    const std::vector<std::string> & interfaces) const = 0;
```